src/charm.py

```diff
@@ -122,7 +123,10 @@ def __init__(self, *args, **kargs) -> None:
                 class.
         """
         super().__init__(*args, **kargs)
-
+        if LXD_PROFILE_YAML.exists():
```

Comment:
Question: Couldn't this be checked on config_changed? So the user can change the test-mode value if this error happens.

Reply:
The `test-mode` here is not intended for user to change the charm into test mode. This configuration is to prevent human error that enabling the test mode in the production environment.

By turning on the `test-mode` alone, the charm does nothing, the developer need to build a special test charm that contains certain files to truly enable test mode. But if someone accidentally uploaded the special test charm to charmhub, this configuration will prevent it from running in the production environment.